src/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -282,11 +283,16 @@ See the LICENSE file in the project root for more information.
       <CustomLinkerArg Include="/OUT:&quot;$(NativeBinary)&quot;" Condition="'$(OS)' == 'Windows_NT'" />
       <CustomLinkerArg Include="/DEF:&quot;$(ExportsFile)&quot;" Condition="'$(OS)' == 'Windows_NT' and $(ExportsFile) != ''" />
       <CustomLinkerArg Include="/LIBPATH:&quot;%(AdditionalNativeLibraryDirectories.Identity)&quot;" Condition="'$(OS)' == 'Windows_NT' and '@(AdditionalNativeLibraryDirectories->Count())' &gt; 0" />
-      <CustomLinkerArg Include="-exported_symbols_list &quot;$(ExportsFile)&quot;" Condition="'$(OS)' != 'Windows_NT' and $(ExportsFile) != ''" />
+      <CustomLinkerArg Include="-exported_symbols_list &quot;$(ExportsFile)&quot;" Condition="'$(TargetOS)' == 'OSX' and $(ExportsFile) != ''" />
       <CustomLinkerArg Condition="Exists('$(_Win32ResFile)')" Include="&quot;$(_Win32ResFile)&quot;" />
       <CustomLinkerArg Include="@(LinkerArg)" />
     </ItemGroup>
-
+	<ItemGroup Condition="'$(OS)' != 'Windows_NT' and '$(TargetOS)' != 'OSX' and $(ExportsFile) != ''">
+		<CustomLinkerArg Include="-Wl,--version-script=$(ExportsFile)" />
+		<CustomLinkerArg Include="-Wl,--discard-all" />
```

Comment:
What the `--discard-all` and `--gc-sections` options strip any debug info? In other words, is the binary linked with these options be hard or impossible to debug?

If these options discard stuff that is truly unreachable, should they be enabled always, and not just when `ExportsFile` is set?

Reply:
No. All debug info is still into the ELF file, I've at least try to debug the .so file and work, It went from C# code to C++ code.
Maybe, but I not haven't tested with no .so compilation.
--discard-all delete all local symbols and as we make all the symbols local at #7249, removes all uneded symbols.
--gc-sections removes all dead code, when we are building shared libraries the compiler assumes that dead code is hidden or not referenced.

Reply:
> All debug info is still into the ELF file, I've at least try to debug the .so file and work, It went from C# code to C++ code.

Thank you for confirming this!